### DISCUSSION TITLE
wii controller RVL-003 mapping

### DISCUSSION THREAD

**Discussion by shawn8888** [ndeadly/MissionControl#390] [General]
I am so excited I paired my Wii controller successfully! 
Is there a place that I can easily see the key mapping to different controllers?  Such as Wii with/without the nunchuck?
Thanks!

**Comment by ndeadly** [comment]
I haven't officially documented the mappings anywhere. Most controllers are mapped pretty intuitively with regards to the positional layout of the switch pro controller. (`+`/`-` take priority over `HOME`/`CAPTURE`, and `HOME`/`CAPTURE` can be simulated by button combos `-`+`DPAD_UP`/`-`+`DPAD_DOWN`).

To find the exact mappings, your best bet for now is to take a look at the source code. The wii controller mappings, for example, can be found here
https://github.com/ndeadly/MissionControl/blob/2ce5b3a7dbd8727ded218d60b2b33a6baf82338e/mc_mitm/source/controllers/wii_controller.cpp#L168-L203

and here for the nunchuck
https://github.com/ndeadly/MissionControl/blob/2ce5b3a7dbd8727ded218d60b2b33a6baf82338e/mc_mitm/source/controllers/wii_controller.cpp#L225-L235

You can also use the button tester in the settings menu to check them.

I would accept a PR to the readme if someone wanted to take the time to document all the default button mappings though.

**Comment by shawn8888** [comment]
Thanks! The button tester works!
System Settings -> Controllers and Sensors -> Test Input Devices
I also find that Wii controller doesn't have X and Y button, even with a nunchuck. 
However the re-mapping in system menu works.

One more question, for controllers don't have home button, such as XBOX one, I know MINUS + DPAD_UP is Home. But how can I press and hold Home to open the quick menu (sleep and change volume, etc)? Is it possible?

**Reply by ndeadly** [reply]
Does holding the combo not work? I don't have my switch handy to check, but I don't see why it wouldn't

**Reply by shawn8888** [reply]
Thank you! Holding combo (MINUS + DPAD_UP) works!
